modules/twinkleblock.js

```diff
@@ -1494,6 +1529,40 @@ Twinkle.block.callback.update_form = function twinkleblockCallbackUpdateForm(e,
 	} else {
 		form.reason.value = data.reason || '';
 	}
+
+	// Clear and/or set any partial page or namespace restrictions
+	if (form.pagerestrictions) {
+		var $pageSelect = $(form).find('[name=pagerestrictions]');
+		var $namespaceSelect = $(form).find('[name=namespacerestrictions]');
+
+		// Respect useInitialOptions by clearing data when switching presets
+		// In practice, this will always clear, since no partial presets use it
+		if (!data.useInitialOptions) {
+			$pageSelect.val(null).trigger('change');
+			$namespaceSelect.val(null).trigger('change');
+		}
+
+		// Add any preset options; in practice, just used for prior block settings
+		if (data.restrictions) {
+			if (data.restrictions.pages && !$pageSelect.val().length) {
+				var pages = data.restrictions.pages.map(function(pr) {
+					return pr.title;
+				});
+				// since page restrictions use an ajax source, we
+				// short-circuit that and just add a new option
+				pages.forEach(function(page) {
+					if (!$pageSelect.find("option[value='" + page + "']").length) {
```

Comment:
```suggestion
					if (!$pageSelect.find("option[value='" + $.escapeSelector(page) + "']").length) {
```

Reply:
I couldn't trigger any issues with any titles here, and still can't.  Maybe because these are values already-returned by the API?  Maybe because of select2?

Reply:
It would fail with a page whose title contains a `'`, like `Occam's razor`.

Reply:
Gah, of course!  I don't know why, I always parse `$.escapeSelector` as just "Escape things you select that might cause issues later" not "Escape things you select that will mess up your actual selection."  Done and thanks!